### PR TITLE
Dart roll helper

### PR DESCRIPTION
**Pull request by bkonyi**
Somehow the last PR was closed by something I did... anyway, addressed changes from the last PR and added error checking. PTAL.

### CODE REVIEW

File: tools/dart/dart_roll_helper.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+MAX_GCLIENT_RETRIES = 3
+
+ERROR_GCLIENT_SYNC_FAILED    = 1
+ERROR_BUILD_FAILED           = 2
+ERROR_PKG_FLUTTER_FAILED     = 3
+ERROR_FLUTTER_GALLERY_FAILED = 4
+
+def engine_golden_licenses_path():
+  return os.path.join(ENGINE_HOME, 'flutter/ci/licenses_golden/')
```
Comment (zanderso) [inline tools/dart/dart_roll_helper.py]: os.path.join(ENGINE_HOME, 'flutter', 'ci', 'licenses_golden') here and below

File: tools/dart/dart_roll_helper.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
```
Comment (zanderso) [inline tools/dart/dart_roll_helper.py]: Two lines between top-level function definitions: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#35-blank-lines

File: tools/dart/dart_roll_helper.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+  return os.path.join(FLUTTER_HOME, 'packages/flutter')
+
+def update_dart_deps_path():
+  return os.path.join(ENGINE_HOME, 'tools/dart/create_updated_flutter_deps.py')
+
+DART_REVISION_ENTRY = 'dart_revision'
+FLUTTER_RUN = ['flutter', 'run']
+FLUTTER_TEST = ['flutter', 'test']
+
+original_revision = ''
```
Comment (zanderso) [inline tools/dart/dart_roll_helper.py]: Avoid global variables: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#254-decision

File: tools/dart/dart_roll_helper.py (hunk 1 above)
Comment (bkonyi) [reply tools/dart/dart_roll_helper.py]: Done.

File: tools/dart/dart_roll_helper.py (hunk 2 above)
Comment (bkonyi) [reply tools/dart/dart_roll_helper.py]: Done.

File: tools/dart/dart_roll_helper.py (hunk 3 above)
Comment (bkonyi) [reply tools/dart/dart_roll_helper.py]: Done (other than ENGINE_HOME, FLUTTER_HOME, and DART_SDK_HOME which are updated in `update_roots`).

**Pull request merged**